src/plugins/intel_gna/src/scale_factor_helper.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "scale_factor_helper.hpp"
+
+#include "common/numerical_utils.hpp"
+#include "gna_plugin_config.hpp"
+#include "log/log.hpp"
+
+namespace ov {
+namespace intela_gna {
+using namespace common;
+namespace helpers {
+
+static bool IsCustomInputScaleFactorAvailableLegacy(const std::vector<float>& input_scale_factors) {
+    if (input_scale_factors.empty()) {
+        return false;
+    }
+
+    bool is_scale_factor_custom = false;
+    for (const auto& scale_factor : input_scale_factors) {
+        if (!fp32eq(scale_factor, GNAPluginNS::kScaleFactorDefault)) {
+            is_scale_factor_custom = true;
+            break;
+        }
+    }
+
+    return is_scale_factor_custom;
+}
+
+static void ApplyScaleFactorsLegacy(const std::vector<float>& input_scale_factors, GNAPluginNS::GnaInputs& inputs) {
+    if (input_scale_factors.size() > inputs.size()) {
+        IE_THROW() << "Configuration input scale factors count is bigger than inputs count";
+    }
+
+    for (size_t id = 0; id < input_scale_factors.size(); ++id) {
+        ::log::warning() << "Using input scale factor: " << input_scale_factors[id]
+                         << ", defined in configuration for input id: " << id << std::endl;
+        inputs.Get().at(id).scale_factor = input_scale_factors[id];
+    }
+}
+
+static bool IsCustomInputScaleFactorPerInputAvailable(const std::map<std::string, float>& per_input_scale_factors) {
+    if (per_input_scale_factors.empty()) {
+        return false;
+    }
+
+    bool is_scale_factor_custom = false;
+    for (const auto& scale_factor : per_input_scale_factors) {
+        if (!fp32eq(scale_factor.second, GNAPluginNS::kScaleFactorDefault)) {
+            is_scale_factor_custom = true;
+            break;
+        }
+    }
+
+    return is_scale_factor_custom;
+}
+
+static void ApplyScaleFactorsPerInput(const std::map<std::string, float>& per_input_scale_factors,
+                                      GNAPluginNS::GnaInputs& inputs) {
+    if (per_input_scale_factors.size() > inputs.size()) {
+        IE_THROW() << "Configuration per input scale factors count is bigger than inputs count";
+    }
+
+    for (auto&& sf : per_input_scale_factors) {
+        auto input_it = inputs.find(sf.first);
+        if (input_it == inputs.end()) {
+            IE_THROW() << "Given scale factor for invalid input: " << sf.first;
+        }
+        ::log::warning() << "Using input scale factor: " << sf.second
+                         << ", defined in configuration for input: " << sf.first << std::endl;
+        input_it->scale_factor = sf.second;
+    }
+}
+
+static bool CheckIfCanApplyCustomScaleFactor(const GNAPluginNS::HeaderLatest::ModelHeader& header) {
+    static constexpr GNAPluginNS::Header2dot8::ModelHeader::Version sc_forbid_overite_scale_factor;
```

Comment:
override

Reply:
It is not clear for me, could you specify a bit more?

Reply:
typo mistake - sc_forbid_overite_scale_factor => sc_forbid_override_scale_factor